TCAT/Controllers/HomeMapViewController.swift

```diff
@@ -0,0 +1,251 @@
+    let minZoom: Float = 12
+    let defaultZoom: Float = 15.5
+    let maxZoom: Float = 25
+    let loadingIndicatorSize = CGSize.init(width: 40, height: 40)
+    
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return banner != nil ? .lightContent : .default
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupMapView()
+        setupOptionsCard()
+
+        setupConstraints()
+    }
+    
+    @objc func reachabilityChanged(_ notification: Notification) {
+        guard let reachability = notification.object as? Reachability else {
+            return
+        }
+        
+        // Dismiss current banner or loading indicator, if any
+        banner?.dismiss()
+        banner = nil
+        
+        delegate?.reachabilityChanged(connection: reachability.connection)
+        
+        switch reachability.connection {
+        case .none:
+            banner = StatusBarNotificationBanner(title: Constants.Banner.noInternetConnection, style: .danger)
+            banner?.autoDismiss = false
+            banner?.show(queuePosition: .front, on: navigationController)
+        default: break
+        }
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        navigationController?.isNavigationBarHidden = true
+        
+        // Add Notification Observers
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(reachabilityChanged(_:)),
+                                               name: .reachabilityChanged,
+                                               object: reachability)
+        do {
+            try reachability?.startNotifier()
+        } catch {
+            print("HomeVC viewDidLayoutSubviews: Could not start reachability notifier")
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        locationManager.delegate = self
+        locationManager.requestWhenInUseAuthorization()
+        
+        StoreReviewHelper.checkAndAskForReview()
+        
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        navigationController?.isNavigationBarHidden = false
+        reachability?.stopNotifier()
+        
+        // Remove Notification Observers
+        NotificationCenter.default.removeObserver(self, name: .reachabilityChanged, object: reachability)
+        
+        // Remove banner and loading indicator
+        banner?.dismiss()
+        banner = nil
+    }
+    
+    func setupMapView() {
+        // Set mapView with settings
+        let camera = GMSCameraPosition.camera(withLatitude: 42.446179, longitude: -76.485070, zoom: defaultZoom)
+        let mapView = GMSMapView.map(withFrame: .zero, camera: camera)
+        mapView.delegate = self
+        mapView.isMyLocationEnabled = true
+        mapView.paddingAdjustmentBehavior = .never // handled by code
```

Comment:
what does `// handled by code` mean? @kevinchan159 help

Reply:
We manually adjust the padding to allow for our own customization. I can look into whether or not we necessarily need to turn this attribute to `.never`